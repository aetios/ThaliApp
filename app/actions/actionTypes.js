export const NAVIGATE = 'NAVIGATE';
export const BACK = 'BACK';
export const OPENDRAWER = 'OPENDRAWER';
export const CALENDARRETREIVED = 'CALENDARRETREIVED';
export const CALENDARERROR = 'CALENDARERROR';
export const RESETLOGINSTATE = 'RESETLOGINSTATE';
export const WELCOME = 'WELCOME';
