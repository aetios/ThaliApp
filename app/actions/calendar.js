/**
 * Created by aucke on 3/8/17.
 */
