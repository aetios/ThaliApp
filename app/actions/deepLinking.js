export const DEEPLINK = 'DEEPLINKING_DEEPLINK';

export function deepLink(url, stayInApp = true) {
  return {
    type: DEEPLINK,
    payload: { url, stayInApp },
  };
}
