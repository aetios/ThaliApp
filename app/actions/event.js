export const EVENT = 'EVENT_EVENT';
export const FETCHING = 'EVENT_FETCHING';
export const SUCCESS = 'EVENT_SUCCESS';
export const FAILURE = 'EVENT_FAILURE';

export function event(pk, token) {
  return {
    type: EVENT,
    payload: { pk, token },
  };
}

export function success(eventData, eventRegistrations) {
  return {
    type: SUCCESS,
    payload: { eventData, eventRegistrations },
  };
}

export function fetching() {
  return {
    type: FETCHING,
  };
}

export function failure() {
  return {
    type: FAILURE,
  };
}
