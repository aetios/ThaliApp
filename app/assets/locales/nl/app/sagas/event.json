{
  "Event added to calendar!": "Evenement aan kalender toegevoegd!",
  "Failed to add event to calendar!": "Evenement toevoegen mislukt!"
}
