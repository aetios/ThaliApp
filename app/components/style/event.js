import { Dimensions } from 'react-native';
import { StyleSheet, colors } from '../../style';

const windowWidth = Dimensions.get('window').width;
export const memberSize = (windowWidth - 64) / 3;

const styles = StyleSheet.create({
  eventView: {
    padding: 16,
    backgroundColor: colors.background,
  },
  locationImageWrapper: {
    marginLeft: -16,
    marginRight: -16,
    marginTop: -16,
    marginBottom: 24,
  },
  locationImage: {
    height: 150,
  },
  titleText: {
    marginBottom: 16,
    fontSize: 20,
    color: colors.darkGrey,
    android: {
      fontFamily: 'sans-serif-medium',
    },
    ios: {
      fontFamily: 'System',
      fontWeight: '600',
    },
  },
  infoHolder: {
    flexDirection: 'row',
  },
  infoText: {
    flex: 1,
    fontSize: 14,
    lineHeight: 22.0,
    color: colors.darkGrey,
    width: '50%',
    paddingRight: 8,
    android: {
      fontFamily: 'sans-serif-medium',
    },
    ios: {
      fontFamily: 'System',
      fontWeight: '600',
    },
  },
  infoValueText: {
    flex: 1,
    fontSize: 14,
    lineHeight: 22.0,
    color: colors.darkGrey,
    width: '50%',
    android: {
      fontFamily: 'sans-serif',
    },
    ios: {
      fontFamily: 'System',
      fontWeight: '300',
    },
  },
  pizzaHolder: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: 16,
  },
  pizzaText: {
    fontSize: 14,
    lineHeight: 22.0,
    color: colors.darkGrey,
    width: '50%',
    paddingRight: 8,
    android: {
      fontFamily: 'sans-serif-medium',
    },
    ios: {
      fontFamily: 'System',
      fontWeight: '600',
    },
  },
  pizzaButton: {
    flex: 1,
  },
  registrationText: {
    marginTop: 16,
  },
  registrationActions: {
    flexDirection: 'row',
    alignItems: 'flex-start',
    marginTop: 16,
  },
  secondButtonMargin: {
    marginLeft: 16,
  },
  descText: {
    fontSize: 14,
    lineHeight: 24.0,
    color: colors.black,
  },
  registrationsTitle: {
    android: {
      fontFamily: 'sans-serif-medium',
    },
    ios: {
      fontFamily: 'System',
      fontWeight: '600',
    },
    fontSize: 14,
    color: colors.darkGrey,
    marginBottom: 16,
  },
  divider: {
    marginLeft: -16,
    marginRight: -16,
    marginTop: 16,
    marginBottom: 16,
    alignSelf: 'stretch',
    borderStyle: 'solid',
    borderWidth: 0.5,
    borderColor: colors.dividerGrey,
  },
  italicText: {
    fontStyle: 'italic',
  },
  flex: {
    flex: 1,
  },
  memberView: {
    marginBottom: 16,
    marginRight: 16,
  },
});

export default styles;
