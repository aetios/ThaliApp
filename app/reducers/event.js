import * as types from '../actions/event';

const initialState = {
  data: {
    pk: -1,
    title: '',
    description: '',
    start: '',
    end: '',
    organiser: -1,
    category: '',
    location: '',
    status: -1,
    registered: false,
  },
  registrations: [],
  success: false,
  hasLoaded: false,
};

export default function loadEvent(state = initialState, action = {}) {
  switch (action.type) {
    case types.FETCHING: {
      return {
        ...state,
        hasLoaded: false,
      };
    }
    case types.SUCCESS:
      return {
        ...state,
        data: action.payload.eventData,
        registrations: action.payload.eventRegistrations,
        success: true,
        hasLoaded: true,
      };
    case types.FAILURE:
      return {
        ...state,
        success: false,
        hasLoaded: true,
      };
    default:
      return state;
  }
}
