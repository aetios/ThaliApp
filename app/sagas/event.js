import {
  call, put, select, takeEvery,
} from 'redux-saga/effects';
import { Sentry } from 'react-native-sentry';

import { apiRequest } from '../utils/url';
import * as eventActions from '../actions/event';
import { tokenSelector } from '../selectors/session';

function* event(action) {
  const { pk, navigateToEventScreen } = action.payload;
  const token = yield select(tokenSelector);

  yield put(eventActions.fetching());

  if (navigateToEventScreen) {
    yield put(eventActions.open());
  }

  const data = {
    method: 'GET',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Token ${token}`,
    },
  };

  try {
    const eventData = yield call(apiRequest, `events/${pk}`, data);

    const params = {
      status: 'registered',
    };

    const eventRegistrations = yield call(apiRequest, `events/${pk}/registrations`, data, params);

    yield put(eventActions.success(
      eventData,
      eventRegistrations,
    ));
  } catch (error) {
    Sentry.captureException(error);
    yield put(eventActions.failure());
  }
}

function* updateRegistration(action) {
  const { pk, present, payment } = action.payload;
  const token = yield select(tokenSelector);

  yield put(eventActions.fetching());

  const data = {
    method: 'PATCH',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Token ${token}`,
    },
    body: JSON.stringify({
      present,
      payment,
    }),
  };

  try {
    yield call(apiRequest, `registrations/${pk}`, data);

    yield put(eventActions.done());
  } catch (error) {
    Sentry.captureException(error);
    yield put(eventActions.failure());
  }
}

function* eventSaga() {
  yield takeEvery(eventActions.EVENT, event);
  yield takeEvery(eventActions.UPDATE_REGISTRATION, updateRegistration);
}

export default eventSaga;
