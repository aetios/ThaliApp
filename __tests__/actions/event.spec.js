import * as actions from '../../app/actions/event';

describe('event actions', () => {
  it('should expose the event actions', () => {
    expect(actions.SUCCESS).toEqual('EVENT_SUCCESS');
    expect(actions.FETCHING).toEqual('EVENT_FETCHING');
    expect(actions.FAILURE).toEqual('EVENT_FAILURE');
    expect(actions.EVENT).toEqual('EVENT_EVENT');
    expect(actions.DONE).toEqual('EVENT_DONE');
  });

  it('should create an action to load an event', () => {
    expect(actions.event(1)).toMatchSnapshot();
  });

  it('should create an action to notify of that an event is fetching', () => {
    expect(actions.fetching()).toMatchSnapshot();
  });

  it('should create an action to notify of a failure fetching an event', () => {
    expect(actions.failure()).toMatchSnapshot();
  });

  it('should create an action to notify of a success fetch of an event', () => {
    expect(actions.success('eventData', 'eventReg')).toMatchSnapshot();
  });

  it('should create an action to notify of a completed fetch of an event', () => {
    expect(actions.done()).toMatchSnapshot();
  });
});