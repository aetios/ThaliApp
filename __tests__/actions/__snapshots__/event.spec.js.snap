// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`event actions should create an action to load an event 1`] = `
Object {
  "payload": Object {
    "navigateToEventScreen": true,
    "pk": 1,
  },
  "type": "EVENT_EVENT",
}
`;

exports[`event actions should create an action to load an event without opening the event screen 1`] = `
Object {
  "payload": Object {
    "navigateToEventScreen": false,
    "pk": 1,
  },
  "type": "EVENT_EVENT",
}
`;

exports[`event actions should create an action to notify of a completed fetch of an event 1`] = `
Object {
  "type": "EVENT_DONE",
}
`;

exports[`event actions should create an action to notify of a failure fetching an event 1`] = `
Object {
  "type": "EVENT_FAILURE",
}
`;

exports[`event actions should create an action to notify of a success fetch of an event 1`] = `
Object {
  "payload": Object {
    "eventData": "eventData",
    "eventRegistrations": "eventReg",
  },
  "type": "EVENT_SUCCESS",
}
`;

exports[`event actions should create an action to notify of that an event is fetching 1`] = `
Object {
  "type": "EVENT_FETCHING",
}
`;

exports[`event actions should create an action to open the admin screen of the event 1`] = `
Object {
  "type": "EVENT_ADMIN",
}
`;

exports[`event actions should create an action to open the event screen 1`] = `
Object {
  "type": "EVENT_OPEN",
}
`;

exports[`event actions should create an action to update a registration 1`] = `
Object {
  "payload": Object {
    "payment": "no_payment",
    "pk": 1,
    "present": true,
  },
  "type": "EVENT_UPDATE_REGISTRATION",
}
`;
